### PR TITLE
[Collision] Apply new factory registration mechanism

### PR DESCRIPTION
**Pull request by fredroy**
See
- #4429

[ci-depends-on https://github.com/SofaDefrost/SoftRobots/pull/306]

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][with-all-tests][force-full-build]

**Comment by fredroy** [comment]
[ci-build][with-all-tests][force-full-build]

**Comment by fredroy** [comment]
[ci-build][with-all-tests][force-full-build]

**Comment by fredroy** [comment]
[ci-build][with-all-tests][force-full-build]

File: examples/Demos/caduceus.scn [hunk 1]
```diff
@@ -35,9 +35,6 @@
     <CollisionResponse name="Response" response="FrictionContactConstraint" />
 
     <InteractiveCamera position="0 30 90" lookAt="0 30 0"/>
-    <LightManager />
-    <SpotLight name="light1" color="1 1 1" position="0 80 25" direction="0 -1 -0.8" cutoff="30" exponent="1" />
-    <SpotLight name="light2" color="1 1 1" position="0 40 100" direction="0 0 -1" cutoff="30" exponent="1" />
```
Comment (bakpaul) [inline examples/Demos/caduceus.scn]: Why ?

Comment (fredroy) [reply examples/Demos/caduceus.scn]: Collateral damage 🫢

**Review by epernod — APPROVED** [review]

**Pull request merged**